### PR TITLE
fix: meaningful error messages for domain validation

### LINKED ISSUE #6550: Error "Domain verification failed" is confusing
### Preflight Checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe your problem

When I want to verify a domain and click on the button 'Verify', then an error message pops-up "Domain verification failed (ORG-GH3s)". From that message it is not clear if the process didn't work or if the DNS records did not match (but the request was ok).
I've checked this and actually the logs return a 200 status code on the request. So I assume it is the DNS record that is the problem. 

Screenshot 2023-09-13 at 16 51 37

### Describe your ideal solution

This message should be improved to make clear what the actual issue is. And if the issue is that the DNS record is missing, the error should look less like a system error and should state this problem clearly to the user so they know what to do.

### Version

_No response_

### Environment

ZITADEL Cloud

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by doncicuto**
In #6550 @mffap stated that 

> When I want to verify a domain and click on the button 'Verify', then an error message pops-up "Domain verification failed (ORG-GH3s). From that message it is not clear if the process didn't work or if the DNS records did not match (but the request was ok). [...] This message should be improved to make clear what the actual issue is. And if the issue is that the DNS record is missing, the error should look less like a system error and should state this problem clearly to the user so they know what to do.

This PR has added some checks to ValidateDomainDNS and ValidateDomainHTTP:

- If the TXT record for _zitadel-challenge is not found the message 'The _zitadel-challenge TXT record was not found for your domain. Check that you've added it to your DNS server or wait till the new record is propagated' is shown

Captura desde 2023-10-08 13-05-02

- If the TXT record for _zitadel-challenge is found but the token text doesn't match the message 'The _zitadel-challenge TXT record has been found for your domain but it doesn't contain the right token text. Check that you've added the right token to your DNS server or wait till the new record is propagated' is shown.

Captura desde 2023-10-08 13-08-18

- If a DNS request has a timeout the message 'There was a timeout querying the DNS server' is shown.
- If the file containing the HTTP is not found (404 Error) the message 'The file containing the challenge was not found in the expected URL. Check that you've uploaded the file in the right place with read permissions' is shown
- If the file containing the HTTP is found but the content doesn't match (quite improbable I know) the message 'The file containing the challenge has been found in the expected URL but it doesn't contain the right token text. Check its content' is shown.

Captura desde 2023-10-08 13-22-40

- Also if the challenge is not found the error erros.ThrowNotFound function is used instead of errors.ThrowInternal as it's less a Zitadel problem that an issue with the record or challenge file.

Closes #6550

### Definition of Ready

- [X] I am happy with the code
- [X] Short description of the feature/issue is added in the pr description
- [X] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [X] No debug or dead code
- [X] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [X] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: internal/static/i18n/it.yaml [hunk 1]
```diff
@@ -182,6 +182,11 @@ Errors:
     DomainVerificationTypeInvalid: Il tipo di verifica del dominio non è valido
     DomainVerificationMissing: La verifica del dominio non è ancora iniziata
     DomainVerificationFailed: Verifica del dominio fallita
+    DomainVerificationTXTNotFound: Il record TXT _zitadel-challenge non è stato trovato per il tuo dominio. Verifica di averlo aggiunto al tuo server DNS o attendi la propagazione del nuovo record
+    DomainVerificationTXTNoMatch: È stato trovato il record TXT _zitadel-challenge per il tuo dominio ma non contiene il testo del token corretto. Verifica di aver aggiunto il token corretto al tuo server DNS o attendi la propagazione del nuovo record
+    DomainVerificationHTTPNotFound: Il file contenente la sfida non è stato trovato nell'URL previsto. Verifica di aver caricato il file nel posto giusto con i permessi di lettura
```
Comment (peintnermax) [inline internal/static/i18n/it.yaml:187]:
```suggestion
    DomainVerificationHTTPNotFound: Il file della challenge non è stato trovato nell'URL previsto. Verifica di aver caricato il file nel posto giusto con permessi di lettura
```

File: internal/static/i18n/it.yaml [hunk 2]
```diff
@@ -182,6 +182,11 @@ Errors:
...
+    DomainVerificationHTTPNoMatch: Il file contenente la sfida è stato trovato nell'URL previsto ma non contiene il testo del token corretto. Controlla il suo contenuto
```
Comment (peintnermax) [inline internal/static/i18n/it.yaml:188]:
```suggestion
    DomainVerificationHTTPNoMatch: Il file della challenge è stato trovato nell'URL previsto ma non contiene il testo del token corretto. Controlla il suo contenuto
```

File: internal/static/i18n/it.yaml [hunk 3]
```diff
@@ -182,6 +182,11 @@ Errors:
...
+    DomainVerificationTimeout: Si è verificato un timeout durante l'interrogazione del server DNS
```
Comment (peintnermax) [inline internal/static/i18n/it.yaml:189]:
```suggestion
    DomainVerificationTimeout: Si è verificato un timeout nella richiesta del server DNS
```

File: internal/static/i18n/de.yaml [hunk 4]
```diff
@@ -182,6 +182,11 @@ Errors:
     DomainVerificationTypeInvalid: Verifikationstyp der Domäne ist ungültig
     DomainVerificationMissing: Verifikation der Domäne noch nicht erstellt
     DomainVerificationFailed: Verifikation der Domäne ist fehlgeschlagen
+    DomainVerificationTXTNotFound: Der TXT-Eintrag _zitadel-challenge wurde für Ihre Domain nicht gefunden. Überprüfen Sie, ob Sie ihn zu Ihrem DNS-Server hinzugefügt haben, oder warten Sie, bis der neue Eintrag verbreitet wird
+    DomainVerificationTXTNoMatch: Der TXT-Eintrag _zitadel-challenge wurde für Ihre Domain gefunden, enthält jedoch nicht den richtigen Token-Text. Überprüfen Sie, ob Sie das richtige Token zu Ihrem DNS-Server hinzugefügt haben, oder warten Sie, bis der neue Eintrag verbreitet wird
+    DomainVerificationHTTPNotFound: Die Datei mit der Herausforderung wurde unter der erwarteten URL nicht gefunden. Überprüfen Sie, ob Sie die Datei mit Leseberechtigungen an der richtigen Stelle hochgeladen haben
```
Comment (peintnermax) [inline internal/static/i18n/de.yaml:187]:
```suggestion
    DomainVerificationHTTPNotFound: Das File der Challenge wurde unter der erwarteten URL nicht gefunden. Überprüfen Sie, ob Sie die Datei mit Leseberechtigungen an der richtigen Stelle hochgeladen haben
```

File: internal/static/i18n/de.yaml [hunk 5]
```diff
@@ -182,6 +182,11 @@ Errors:
...
+    DomainVerificationHTTPNoMatch: Die Datei mit der Herausforderung wurde in der erwarteten URL gefunden, enthält jedoch nicht den richtigen Token-Text. Überprüfen Sie den Inhalt
```
Comment (peintnermax) [inline internal/static/i18n/de.yaml:188]:
```suggestion
    DomainVerificationHTTPNoMatch: Das File der Challenge wurde in der erwarteten URL gefunden, enthält jedoch nicht den richtigen Token-Text. Überprüfen Sie den Inhalt
```

**Review by eliobischof — APPROVED** [review]
Thank you @doncicuto!
Great work as always :tada: 

IMO, it's more a fix than a feature.

**Pull request merged**